metrics/cli/lib/services/firebase/cli/firebase_cli.dart

```diff
@@ -26,55 +26,101 @@ class FirebaseCli extends Cli {
 
   /// Sets the project with the [projectId] identifier as the default one
   /// for the Firebase project in the [workingDirectory].
-  Future<void> setFirebaseProject(String projectId, String workingDirectory) {
-    return run(['use', projectId], workingDirectory: workingDirectory);
+  ///
+  /// The [authToken] is an optional parameter for the direct
+  /// Firebase authentication.
+  Future<void> setFirebaseProject(
+    String projectId,
+    String workingDirectory, [
+    String authToken,
+  ]) {
+    return run([
+      'use',
+      projectId,
+      if (authToken != null) '--token=$authToken',
+    ], workingDirectory: workingDirectory);
   }
 
   /// Clears the Firebase [target] in the given [workingDirectory].
-  Future<void> clearTarget(String target, String workingDirectory) {
-    return run(
-      ['target:clear', 'hosting', target],
-      workingDirectory: workingDirectory,
-    );
+  ///
+  /// The [authToken] is an optional parameter for the direct
+  /// Firebase authentication.
+  Future<void> clearTarget(
+    String target,
+    String workingDirectory, [
+    String authToken,
+  ]) {
+    return run([
+      'target:clear',
+      'hosting',
+      target,
+      if (authToken != null) '--token=$authToken',
+    ], workingDirectory: workingDirectory);
   }
 
   /// Associates the Firebase [target] with the given [hostingName]
   /// in the given [workingDirectory].
+  ///
+  /// The [authToken] is an optional parameter for the direct
+  /// Firebase authentication.
   Future<void> applyTarget(
     String hostingName,
     String target,
-    String workingDirectory,
-  ) {
-    return run(
-      ['target:apply', 'hosting', target, hostingName],
-      workingDirectory: workingDirectory,
-    );
+    String workingDirectory, [
+    String authToken,
+  ]) {
+    return run([
+      'target:apply',
+      'hosting',
+      target,
+      hostingName,
+      if (authToken != null) '--token=$authToken',
+    ], workingDirectory: workingDirectory);
   }
 
   /// Deploys a project's [target] from the given [workingDirectory]
   /// to the Firebase hosting.
-  Future<void> deployHosting(String target, String workingDirectory) {
-    return run(
-      ['deploy', '--only', 'hosting:$target'],
-      workingDirectory: workingDirectory,
-    );
+  ///
+  /// The [authToken] is an optional parameter for the direct
+  /// Firebase authentication.
+  Future<void> deployHosting(
+    String target,
+    String workingDirectory, [
+    String authToken,
+  ]) {
+    return run([
+      'deploy',
+      '--only',
+      'hosting:$target',
+      if (authToken != null) '--token=$authToken',
```

Comment:
:umbrella_with_rain_drops:
What do you think about extracting the creation of a list of arguments into a separate method to avoid duplication of the following line: `if (authToken != null) '--token=$authToken'`? Or maybe it makes sense to create separate wrapper classes for each CLI command that will contain all possible arguments like it is done in the [driver tests](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/test_driver/cli/flutter/command/drive_command.dart) script? What do you think?